src/docfx/build/dependency/DependencyResolver.cs

```diff
@@ -318,7 +305,55 @@ private Document TryResolveFile(Document referencingFile, string pathToDocset)
             return default;
         }
 
-        private Document TryResolveResourceFromHistory(GitCommitProvider gitCommitProvider, string pathToDocset, TemplateEngine templateEngine)
+        private Document CreateFromGit(Docset docset, FilePath filePath, bool deleted = false)
+        {
+            Debug.Assert(_gitCommitProvider != null);
+            Debug.Assert(filePath != null);
+
+            if (docset == null)
+            {
+                return default;
+            }
+
+            if (File.Exists(Path.Combine(docset.DocsetPath, filePath.Path)))
+            {
+                return Document.Create(docset, filePath, _templateEngine);
+            }
+
+            var (repo, pathToRepo, commits) = _gitCommitProvider.GetCommitHistory(docset, filePath.Path, docset.Repository.Commit);
+
+            var commit = deleted && commits.Length > 1 ? commits[1] : (!deleted && commits.Length > 0 ? commits[0] : default);
```

Comment:
GetCommitHistory cannot tell you if a file exist in the HEAD commit, this can potentially resolve a deleted dependency file

Reply:
That's what I met in the integration test, will find a way to know that.